apis/datadoghq/v2alpha1/datadogagent_types.go

```diff
@@ -354,6 +354,12 @@ type DogstatsdFeatureConfig struct {
 	// +optional
 	OriginDetectionEnabled *bool `json:"originDetectionEnabled,omitempty"`
 
+	// TagCardinality configures tag cardinality for the metrics collected using origin detection (`low`, `orchestrator` or `high`).
+	// See also: https://docs.datadoghq.com/getting_started/tagging/assigning_tags/?tab=containerizedenvironments#environment-variables
+	// Cardinality default: low
+	// +optional
+	TagCardinality *string `json:"tagCardinality,omitempty"`
```

Comment:
in lieu of the function below to check for validity, we could add a new struct here `TagCardinalityName` (for example), and then three constants of this type - similar to what we do for `AgentConfigFileName`

Reply:
Addressed in comment below